src/DotNetWorker.Core/FunctionsApplication.cs

```diff
@@ -61,12 +65,24 @@ public void LoadFunction(FunctionDefinition definition)
             _diagnostics.OnFunctionLoaded(definition);
         }
 
-        public Task InvokeFunctionAsync(FunctionContext context)
+        public async Task InvokeFunctionAsync(FunctionContext context)
         {
             var scope = new FunctionInvocationScope(context.FunctionDefinition.Name, context.InvocationId);
-            using (_logger.BeginScope(scope))
+
+            using var logScope = _logger.BeginScope(scope);
+            using Activity? invokeActivity = _functionActivitySourceFactory.StartInvoke(context);
+
+            try
             {
-                return _functionExecutionDelegate(context);
+                await _functionExecutionDelegate(context);
+                invokeActivity?.SetStatus(ActivityStatusCode.Ok);
+            }
+            catch (Exception ex)
+            {
+                invokeActivity?.SetStatus(ActivityStatusCode.Error, ex.Message);
+                invokeActivity?.RecordException(ex);
```

Comment:
Do you think we should also set exception.escapd? https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/exceptions/#:~:text=Requirement%20Level-,exception.escaped,-boolean

Reply:
Yup!